t/class/class.t

```diff
@@ -0,0 +1,86 @@
+#!./perl
+
+BEGIN {
+    chdir 't' if -d 't';
+    require './test.pl';
+    set_up_inc('../lib');
+    require Config;
+}
+
+use v5.36;
+use feature 'class';
+no warnings 'experimental::class';
+
+{
+    class Test1 {
+        method hello { return "hello, world"; }
+    }
+
+    my $obj = Test1->new;
+    isa_ok($obj, "Test1", '$obj');
+
+    is($obj->hello, "hello, world", '$obj->hello');
+}
+
+# Classes are still regular packages
+{
+    class Test2 {
+        my $ok = "OK";
+        sub NotAMethod { return $ok }
```

Comment:
Will it (having `my` and `sub` within classes) go away in the future?

What's the use of it? Private/static methods will have `method :private` syntax I believe.

It's currently a bit confusing.

Reply:
@rwp0 Not sure if I understand your question, but I'll try to answer what I think you're asking.

`sub` will still be allowed in classes. There's no problem with that, but eventually we'll need work to ensure that subroutines cannot be called as methods.

Also, when the MOP is online, we'll need debugger work to ensure that typing `m` only shows subroutines, not methods (might be possibly now, but I haven't looked into it).

`my` will also be allowed in a top-level scope in a `class` (no sense arbitrarily disallowing it). It will effectively be class data, but as a `my` variable would be an internal implementation detail rather than a structural part of the class, its use should be _strongly_ discouraged if the anything outside the class is able to use that data.